### PR TITLE
issue #57: Async await tests

### PR DESCRIPTION
**Pull request by sogoiii**
Transformed all tests to async await pattern. 

Tests are now consistent ---> https://github.com/OpenZeppelin/zeppelin-solidity/issues/57

### CODE REVIEW

File: test/BasicToken.js [hunk 1]
```diff
@@ -1,49 +1,32 @@
...
-      })
-      .then(function(totalSupply) {
-        assert.equal(totalSupply, 100);
-      })
-      .then(done);
+  it("should return the correct totalSupply after construction", async function(done) {
+    let token = await BasicTokenMock.new(accounts[0], 100);
+    let totalSupply = await token.totalSupply();
+    assert.equal(totalSupply, 100);
+    done();
```
Comment (federicobond) [inline test/BasicToken.js]: No need to call `done` here, and we can remove it from the parameter too.

File: test/BasicToken.js [hunk 2]
```diff
@@ -1,49 +1,32 @@
...
-      })
-      .then(done);
+  it("should return correct balances after transfer", async function(done){
+    let token = await BasicTokenMock.new(accounts[0], 100);
+    let transfer = await token.transfer(accounts[1], 100);
+    let firstAccountBalance = await token.balanceOf(accounts[0]);
+    assert.equal(firstAccountBalance, 0);
+    let secondAccountBalance = await token.balanceOf(accounts[1]);
+    assert.equal(secondAccountBalance, 100);
+    done();
```
Comment (federicobond) [inline test/BasicToken.js]: Same here with `done`. We can also add an empty line before each let/assert to improve readability.

File: test/BasicToken.js [hunk 3]
```diff
@@ -1,49 +1,32 @@
...
-      })
-      .then(done);
+  it("should throw an error when trying to transfer more than balance", async function(done) {
+
+    let token = await BasicTokenMock.new(accounts[0], 100);
+    try {
+      let transfer = await token.transfer(accounts[1], 101);
+    } catch(error) {
+      if (error.message.search('invalid JUMP') === -1) throw error
+      assert.isAbove(error.message.search('invalid JUMP'), -1, 'Invalid JUMP error must be returned');
```
Comment (federicobond) [inline test/BasicToken.js]: We should write a quick helper for asserting throws, as it will be used many times.

File: test/BasicToken.js (hunk 1 above)
Comment (sogoiii) [inline test/BasicToken.js]: I have 2 or 3 tests that require the done. I basically went with consistency. Ill remove the done from all the other tests.

Comment (federicobond) [inline test/BasicToken.js]: Can you point me towards these tests? Mention me on the appropriate line.

File: test/Bounty.js [hunk 4]
```diff
@@ -8,134 +8,108 @@ var sendReward = function(sender, receiver, value){
...
-          return bounty.checkInvariant.call()
-        }).
-        then(function(result) {
-          assert.isFalse(result);
-        }).
-        then(done);
+    it("checkInvariant returns false", async function(){
+      let bounty = await InsecureTargetBounty.new();
+      let target = await bounty.createTarget();
+      let invarriantCall = await bounty.checkInvariant.call();
```
Comment (federicobond) [inline test/Bounty.js]: `invariantCall`

File: test/BasicToken.js (hunk 1 above)
Comment (sogoiii) [inline test/BasicToken.js]: Its the ones in bounty.

Comment (federicobond) [inline test/BasicToken.js]: Yes, it's fine to use `done` when you have to listen to events.

**Comment by federicobond** [comment]
@sogoiii there is a failing test case for `PullPayment can withdraw payment`. I could reproduce it locally but could not find the cause. Do you have any ideas?

**Comment by maraoz** [comment]
All tests passed locally for me. Weird. Is any test non-deterministic?

**Comment by sogoiii** [comment]
@federicobond I was on node 6.3.1, which makes that error go away. I rememeber as i was doing this, i found an issue that suggested the downgrade. I currently cannot find that issue. But that was why i changed my node version.  I simply cannot find that github issue.

**Comment by sogoiii** [comment]
Hmm... I Dont know the combination of testrpc and node to make this work.

**Comment by sogoiii** [comment]
@federicobond can you try truffle test instead of npm test?

**Comment by federicobond** [comment]
Yes, it works with my globally installed `truffle`. A dependencies issue, maybe?

**Comment by sogoiii** [comment]
Yea, ill keep looking for the problem. Glad im not crazy about that npm and truffle difference.

**Comment by sogoiii** [comment]
@federicobond @maraoz Looks like if i change the version of solc from 0.4.4 to either 0.4.5 or 0.4.6, the error appears. 

Looking at their repo, the only thing they added between 0.4.4 and 0.4.5 is this [commit](https://github.com/ethereum/solc-js/pull/6)

Just to be clear on my procedure. On my global truffle folder, i hardcoded the 0.4.4 and ran a successful `truffle test`. I then hard coded the package to 0.4.5, npm installed, and ran `truffle test`, only to get the error.

**Comment by maraoz** [comment]
I created this branch to see if travis was failing on master: https://github.com/OpenZeppelin/zeppelin-solidity/pull/98

Apparently this problem is unrelated to this PR, so I'll merge. 
Opening an issue to investigate and solve this problem: https://github.com/OpenZeppelin/zeppelin-solidity/issues/99

**Comment by maraoz** [comment]
@sogoiii thanks for the PR and the research on the problem with travis!

**Comment by sogoiii** [comment]
sweet!

**Pull request merged**